source/fuzz/transformation_vector_shuffle.h

```diff
@@ -59,7 +59,8 @@ class TransformationVectorShuffle : public Transformation {
   // from which it came (with undefined components being ignored).  If the
   // result vector is a contiguous sub-range of one of the input vectors, a
   // fact is added to record that |message_.fresh_id| is synonymous with this
-  // sub-range.
+  // sub-range. DataSynonym facts are added only if neither |vector1| nor
+  // |vector2| is an irrelevant id.
```

Comment:
If you act on my suggestion above, we would add data synonym facts for whichever of the two vectors (possibly both, possibly neither) is relevant.

Reply:
Fixed in 5bea7f6.